tlaplus/ghostferry.tla

```diff
@@ -284,12 +347,12 @@ PossibleRecords == Records \cup {NoRecordHere}
       \* It may be possible to perform some sort of locking via the Application,
       \* but this seems cumbersome and prone to implementation level error.
       \* TODO: model this with TLA+ and validate its correctness.
-      tblit_loop:  while (lastSuccessfulPK < MaxPrimaryKey) {
-      tblit_rw:      currentRow := SourceTable[lastSuccessfulPK + 1];
-                     if (currentRow # NoRecordHere) {
-                       TargetTable[lastSuccessfulPK + 1] := currentRow;
+      tblit_loop:  while (LastSuccessfulPK < MaxPrimaryKey) {
+      tblit_rw:      currentRow := SourceTable[LastSuccessfulPK + 1];
+                     if (currentRow # NoRecordHere /\ TargetTable[LastSuccessfulPK + 1] = NoRecordHere) {
```

Comment:
A lot of drama came with this line of change. A very detailed explanation is available at the commit message of: https://github.com/Shopify/ghostferry/pull/207/commits/1daf5fdd393f5bdd240e90ad11100e9148ad4767

Reply:
This took me a while to understand. I think this is only okay because we do `SELECT FOR UPDATE` and `REPLACE INTO` or `INSERT IGNORE` for the data copy as an atomic operation.

Reply:
I think your explanation above is correct. As @xliang6 stated above, this is predicated on the fact that the DataIterator operates atomically and is not subject to a race condition with the BinlogStreamer.